components/skus/browser/skus_context_impl.cc

```diff
@@ -45,27 +45,47 @@ logging::LogSeverity GetLogSeverity(skus::TracingLevel level) {
 
 namespace skus {
 
-FetchOrderCredentialsCallbackState::FetchOrderCredentialsCallbackState() =
-    default;
-FetchOrderCredentialsCallbackState::~FetchOrderCredentialsCallbackState() =
-    default;
-
-PrepareCredentialsPresentationCallbackState::
-    PrepareCredentialsPresentationCallbackState() = default;
-PrepareCredentialsPresentationCallbackState::
-    ~PrepareCredentialsPresentationCallbackState() = default;
+skus::mojom::SkusResultPtr to_skus_result(const skus::SkusResultCode code,
+                                          const std::string& message) {
+  return skus::mojom::SkusResult::New(
+      static_cast<skus::mojom::SkusResultCode>(code), message);
+}
 
-CredentialSummaryCallbackState::CredentialSummaryCallbackState() = default;
-CredentialSummaryCallbackState::~CredentialSummaryCallbackState() = default;
+RustSequencedCallback::RustSequencedCallback(
+    base::OnceCallback<void(skus::mojom::SkusResultPtr)> callback,
```

Comment:
`BindPostTask`

Reply:
and maybe we call the class something like `RustBoundPostTask`?

Reply:
Chromium says to use `BindPostTaskToCurrentDefault` as a last resort and prefers to use an explicit task runner: 
<img width="632" alt="image" src="https://github.com/brave/brave-core/assets/1530031/9ecaefc1-d2e7-4f0c-8164-e76d702b44b0">

Should I still change it?